internal/common/commontest/inputs.go

```diff
@@ -0,0 +1,64 @@
+package commontest
+
+import (
+	"encoding/json"
+	"io"
+	"log"
+	"net/http"
+	"net/url"
+	"path"
+	"strings"
+)
+
+// AddUserInput adds user controllable data to the request r.
+// The data type can be configured with inputType. If inputType is not
+// supported, the program exits.
+// You can also specify the key and value of the data to be added to
+// the request. The key "input" and value "fake-user-input"
+// are used by default.
+func AddUserInput(inputType string, r *http.Request, key, value string) {
+	if key == "" {
+		key = "input"
+	}
+	if value == "" {
+		value = "fake-user-input"
+	}
+	switch inputType {
+	case "query", "buffered-query":
```

Comment:
Do you think these should be constants? If we know we only support specific input types, I feel like the input types can be an enum.

Reply:
I think we can split this out into a separate ticket. Switching to enum constants is a good idea but it requires changes to a number of files and will make this PR more difficult to review IMO.